src/litserve/server.py

```diff
@@ -319,7 +342,7 @@ async def event_wait(evt, timeout):
             def get_from_pipe():
                 if read.poll(self.app.timeout):
                     return read.recv()
-                return HTTPException(status_code=504, detail="Request timed out")
+                raise HTTPException(status_code=504, detail="Request timed out")
```

Comment:
we can just raise the `HTTPException` here and FastAPI will handle things for us.